expertsystem/reaction/conservation_rules.py

```diff
@@ -297,7 +305,7 @@ class GParityNodeInput:
     s_mag: NodeQuantumNumbers.s_magnitude = attr.ib()
 
 
-class GParityConservation(Rule):
+class GParityConservation:
```

Comment:
Wait, why do we still use classes now that there are no abstract base classes anymore? There was some problem with the signatures or so, right?

(Can also be handled in a follow-up issue)

Reply:
This is related to the fact that I was not able to get the Protocols working for regular functions.

Reply:
Since I now know that the full signature of the protocol functions is important, also the argument names, I can verify that it also works with regular functions. So this can be changed